source/data-formats/modeling-bson-data.txt

```diff
@@ -157,62 +215,59 @@ The same document in the MongoDB shell might display as:
 
 .. note::
 
-   :php:`MongoDB\BSON\Persistable <mongodb-bson-persistable>` may only be used
-   for root and embedded BSON documents. It may not be used for BSON arrays.
+   You can use the ``Persistable`` interface for root and embedded BSON documents
+   only, not BSON arrays.
 
-Working with Enums
-------------------
+.. _php-enums:
 
-:php:`Backed enums <enumerations.backed>` can be used with BSON and will
-serialize as their case value (i.e. integer or string).
-:php:`Pure enums <enumerations.basics>`, which have no backed cases, cannot be
-directly serialized. This is similar to how enums are handled by
-:php:`json_encode() <json-encode>`.
+Enum Values
+~~~~~~~~~~~
 
-Round-tripping a backed enum through BSON requires special handling. In the
-following example, the ``bsonUnserialize()`` method in the class containing the
-enum is responsible for converting the value back to an enum case:
+You can serialize and deserialize backed enums into BSON data. Backed 
+enum values serialize as their case value, while pure enums without
+case values cannot be directly serialized. To perform these conversions,
+you must specify serialization logic by defining the ``bsonSerialize()``
+and ``bsonUnserialize()`` methods.
 
-.. code-block:: php
+.. tip::
 
-   <?php
+   To learn more about backed enums, see :php:`Backed enums <enumerations.backed>`
+   in the {+extension-short+} documentation.
 
-   enum Role: int
-   {
-       case USER = 1;
-       case ADMIN = 2;
-   }
+Example
+```````
 
-   class User implements MongoDB\BSON\Persistable
-   {
-       public function __construct(
-           private string $username,
-           private Role $role,
-           private MongoDB\BSON\ObjectId $_id = new MongoDB\BSON\ObjectId(),
-       ) {}
-
-       public function bsonSerialize(): array
-       {
-           return [
-               '_id' => $this->_id,
-               'username' => $this->username,
-               'role' => $this->role,
-           ];
-       }
-
-       public function bsonUnserialize(array $data): void
-       {
-           $this->_id = $data['_id'];
-           $this->username = $data['username'];
-           $this->role = Role::from($data['role']);
-       }
-   }
+The following example specifies logic for serializing and deserializing the
+``Role`` enum by defining the ``bsonSerialize()`` and ``bsonUnserialize()``
+methods:
+
+.. literalinclude:: /includes/bson.php
+    :language: php
+    :dedent:
+    :start-after: start-backed-enum
+    :end-before: end-backed-enum
+
```

Comment:
S: consider demonstrating an example as you did in the previous section of the serialization roundtrip